src/txParsers.ts

```diff
@@ -0,0 +1,282 @@
+    const multiassetMap = parseMap(
+        unparsedMultiasset,
+        parsePolicyId,
+        createParser(
+            parseMap,
+            parseAssetName,
+            parseAssetValue,
+            errMsg,
+        ),
+        errMsg,
+    )
+
+    return Array.from(multiassetMap).map(([policyId, tokens]) => ({
+        policyId,
+        tokens: Array.from(tokens).map(([assetName, amount]) => ({assetName, amount})),
+    }))
+}
+
+const parseTxInput = (unparsedTxInput: unknown): TransactionInput => {
+    const [transactionId, index] = parseTuple(
+        unparsedTxInput,
+        ParseErrorReason.INVALID_TX_INPUTS,
+        createParser(parseBufferOfLength, TX_ID_HASH_LENGTH, ParseErrorReason.INVALID_TRANSACTION_ID),
+        createParser(parseUint, ParseErrorReason.INVALID_TX_INPUT_INDEX),
+    )
+
+    return {transactionId, index}
+}
+
+const parseAmountWithMultiasset = (unparsedAmount: unknown): Amount => {
+    const [coin, multiasset] = parseTuple(
+        unparsedAmount,
+        ParseErrorReason.INVALID_OUTPUT_AMOUNT,
+        createParser(parseUint, ParseErrorReason.INVALID_OUTPUT_AMOUNT),
+        createParser(parseMultiasset, createParser(parseUint, ParseErrorReason.INVALID_OUTPUT_MULTIASSET), ParseErrorReason.INVALID_OUTPUT_MULTIASSET),
+    )
+
+    return {type: AmountType.WITH_MULTIASSET, coin, multiasset}
+}
+
+const parseAmountWithoutMultiasset = (unparsedAmount: unknown): Amount => ({
+    type: AmountType.WITHOUT_MULTIASSET,
+    coin: parseUint(unparsedAmount, ParseErrorReason.INVALID_OUTPUT_AMOUNT),
+})
+
+const parseAmount = (unparsedAmount: unknown): Amount =>
+    isArray(unparsedAmount)
+        ? parseAmountWithMultiasset(unparsedAmount)
+        : parseAmountWithoutMultiasset(unparsedAmount)
+
+const parseTxOutput = (unparsedTxOutput: unknown): TransactionOutput => {
+    const [address, amount] = parseTuple(
+        unparsedTxOutput,
+        ParseErrorReason.INVALID_TX_OUTPUT,
+        createParser(parseBuffer, ParseErrorReason.INVALID_OUTPUT_ADDRESS),
+        parseAmount,
+    )
+
+    return {address, amount}
+}
+
+export const parseWithdrawals = (unparsedWithdrawals: unknown): Withdrawal[] => {
+    const withdrawalsMap = parseMap(unparsedWithdrawals, parseRewardAccount, createParser(parseUint, ParseErrorReason.INVALID_WITHDRAWAL_AMOUNT), ParseErrorReason.INVALID_TX_WITHDRAWALS)
+
+    return Array.from(withdrawalsMap).map(([rewardAccount, amount]) => ({rewardAccount, amount}))
+}
+
+const parseStakeCredentialType = (unparsedStakeCredentialType: unknown): StakeCredentialType => {
+    validate(isNumber(unparsedStakeCredentialType), ParseErrorReason.INVALID_STAKE_CREDENTIAL_TYPE)
+    validate(unparsedStakeCredentialType in StakeCredentialType, ParseErrorReason.INVALID_STAKE_CREDENTIAL_TYPE)
+    return unparsedStakeCredentialType
+}
+
+const parseStakeCredentialAddr = (data: unknown[]): WithoutType<StakeCredentialAddress> => ({
```

Comment:
```suggestion
const parseStakeCredentialKey = (data: unknown[]): WithoutType<StakeCredentialAddress> => ({
```

Reply:
Shouldn't it be more like `parseStakeCredentialAddressKey`?

Reply:
Changed to `parseStakeCredentialKey` as discussed below.